src/Database/LSMTree/Internal/RawPage.hs

```diff
@@ -50,13 +105,121 @@ rawPageNumBlobs :: RawPage -> Word16
 rawPageNumBlobs (RawPage off ba) = indexByteArray ba (off + 1)
 
 type KeyOffset = Word16
+type ValueOffset = Word16
 
-rawPageKeyOffsets :: RawPage -> Vector KeyOffset
+rawPageKeyOffsets :: RawPage -> P.Vector KeyOffset
```

Comment:
Perhaps document here, as a helpful reminder, that the key offsets array has n+1 entries, so we can lookup the span for all n keys.